<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>`BHMSMAfMRI` User Guide • BHMSMAfMRI</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="`BHMSMAfMRI` User Guide">
<meta property="og:description" content="BHMSMAfMRI">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BHMSMAfMRI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BHMSMAfMRIvignette.html">`BHMSMAfMRI` User Guide</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/nilotpalsanyal/BHMSMAfMRI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>
<code>BHMSMAfMRI</code> User Guide</h1>
                        <h4 data-toc-skip class="author">Nilotpal
Sanyal</h4>
            
            <h4 data-toc-skip class="date">September 08, 2022</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/nilotpalsanyal/BHMSMAfMRI/blob/HEAD/vignettes/BHMSMAfMRIvignette.Rmd" class="external-link"><code>vignettes/BHMSMAfMRIvignette.Rmd</code></a></small>
      <div class="hidden name"><code>BHMSMAfMRIvignette.Rmd</code></div>

    </div>

    
    
<!-- \setlength{\textwidth}{6in}
\setlength{\textheight}{8.5in}
\setlength{\parskip}{.2cm plus4mm minus2mm}
 -->
<!-- ```{r set-options, echo=FALSE, cache=FALSE}
options(width = 200)
```
 -->
<p><span class="math inline">\(\hspace{5cm}\)</span></p>
<p>Welcome to <strong>BHMSMAfMRI</strong>, an R package to analyze
functional MRI (fMRI) data, or other multiscale data. This manual shows
how the <strong>BHMSMAfMRI</strong> package functions are used to
analyze fMRI data and should be helpful for the first-time user. In
Section @ref(secintro), we give a short introduction and
non-mathematical overview of the methodology, and in Section
@ref(secuse), we discuss the package functions in a systematic way and
apply them to analyze a simulated fMRI dataset.</p>
<div class="section level2">
<h2 id="secintro">Introduction and overview<a class="anchor" aria-label="anchor" href="#secintro"></a>
</h2>
<p>The <strong>BHMSMAfMRI</strong> package performs Bayesian
hierarchical multi-subject multiscale analysis (BHMSMA) of fMRI data
<span class="citation">(Sanyal and Ferreira 2012)</span>, or other
multiscale data. Though fMRI data is generally 3D, currently
<strong>BHMSMAfMRI</strong> considers analysis of 2D slices only.</p>
<p>The main features of the BHMSMA method are that:</p>
<ul>
<li>it takes into account both the temporal and the spatial information
contained in the fMRI data,</li>
<li>it performs multi-subject analysis and borrows strength across
subjects for precise estimation of the effect sizes (i.e., brain
activations), and provides a straightforward way to obtain group
activation map, and</li>
<li>it does not use Markov Chain Monte Carlo (MCMC) simulation and is
fast.</li>
</ul>
<p>BHMSMA models the temporal variation present in the fMRI data through
a general linear model (GLM) and then considers discrete wavelet
transform of the standardized regression coefficients for harnessing the
spatial information. In the wavelet domain, each wavelet coefficient is
assigned a mixture prior that is a combination of a Gaussian density and
a point mass at zero. This prior specification takes into account the
sparsity of the wavelet coefficients. For the mixture probabilities
BHMSMA considers a prior that depends on few hyperparameters. Inference
is carried out by an empirical Bayes methodology without using MCMC
methods. The inference uses approximation of one-dimensional integrals
only. The posterior mean of the regression coefficients are obtained by
using the posterior mean of the wavelet coefficients in the inverse
discrete wavelet transform. Further, the posterior wavelet coefficients
are averaged over subjects and used in the inverse discrete wavelet
transform to obtain posterior group estimate of the regression
coefficients. Posterior uncertainty is assessed based on simulations
from the posterior distribution of the regression coefficients.</p>
<p>The <strong>BHMSMAfMRI</strong> package fits the BHMSMA model to the
fMRI data and provides estimates of the hyperparameters along with their
standard error, posterior mean of the wavelet coefficients, posterior
mean of the regression coefficients, samples from the posterior
distribution of the regression coefficients and posterior group estimate
of the regression coefficients. The posterior samples can be used to
compute the estimates of posterior standard deviation and posterior
probability maps.</p>
</div>
<div class="section level2">
<h2 id="secuse">Use of package functions with examples<a class="anchor" aria-label="anchor" href="#secuse"></a>
</h2>
<p>In this section we illustrate the use of the package functions. We
assume that prior to applying our methodology, the fMRI data have been
preprocessed for necessary corrections like realignment or motion
correction, slice-timing correction, coregistration with anatomical
image and normalization. However, the data <em>must not be</em>
spatially filtered before applying BHMSMA, because our approach is to
include the spatial information into modeling instead of filtering it
out. Preprocessing can be perfomed by using available softwares/packages
like SPM <span class="citation">(Friston et al. 2007)</span>,
BrainVoyager <span class="citation">(Goebel, Esposito, and Formisano
2006)</span>, AFNI <span class="citation">(Cox 1996)</span>, and FSL
<span class="citation">(Smith et al. 2004)</span>. In the following
subsections, we show the use of the package functions in a systematic
way.</p>
<div class="section level3">
<h3 id="the-main-function-bhmsma">The main function <code>BHMSMA</code><a class="anchor" aria-label="anchor" href="#the-main-function-bhmsma"></a>
</h3>
<p><code>BHMSMA</code> is the main function of the
<strong>BHMSMAfMRI</strong> package. <code>BHMSMA</code> accepts fMRI
data as a 4D array. The data can be imported from various image files by
the <code>readfmridata</code> function. <code>BHMSMA</code> successively
calls the following functions to perform the whole analysis —
<code>glmcoef</code> to obtain regression coefficient map of the
regressor of interest, <code>waveletcoef</code> to perform 2D wavelet
transformation of the regression coefficient map,
<code>hyperparamest</code> to obtain estimates of the BHMSMA model
hyperparameters, <code>postmixprob</code> to obtain estimates of the
mixture probabilities that define posterior distribution of the wavelet
coefficients, <code>postwaveletcoef</code> to obtain the posterior
estimates of the wavelet coefficients and finally
<code>postglmcoef</code> to obtain the posterior smoothed version of the
regression coefficient map. If true coefficients are given,
<code>BHMSMA</code> also returns mean squared error (MSE) estimates.
Here is a quick look at the usage and outputs of the function.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/nilotpalsanyal/BHMSMAfMRI" class="external-link">BHMSMAfMRI</a></span><span class="op">)</span></span>
<span><span class="va">BHMSMAmulti</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/BHMSMA.html">BHMSMA</a></span><span class="op">(</span><span class="va">n</span>, <span class="va">grid</span>, <span class="va">data</span>, <span class="va">designmat</span>, <span class="va">k</span>, <span class="st">"multi"</span>, <span class="va">truecoef</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">BHMSMAmulti</span><span class="op">)</span></span>
<span><span class="co">#[1] "GLMCoefStandardized"      "GLMCoefSE"                "WaveletCoefficientMatrix"</span></span>
<span><span class="co">#[4] "hyperparam"               "hyperparamVar"            "posteriorMixProb"        </span></span>
<span><span class="co">#[7] "Waveletcoefposterior"     "GLMcoefposterior"        </span></span></code></pre></div>
<p>The dimension of <code>data</code> is
<code>c(n,grid,grid,ntime)</code> where <code>n</code> is the number of
sujects, <code>grid^2</code> is the total number of voxels in the data
and <code>ntime</code> is the length of time-series for each voxel. The
argument <code>k</code> selects the regressor of interest from the
design matrix <code>designmat</code> which can have multiple regressor
columns. Following sections break down the whole analysis shown above by
showing specific uses and outputs of all the above functions and several
others of the <strong>BHMSMAfMRI</strong> package.</p>
</div>
<div class="section level3">
<h3 id="reading-fmri-data-readfmridata">Reading fMRI Data: <code>readfmridata</code><a class="anchor" aria-label="anchor" href="#reading-fmri-data-readfmridata"></a>
</h3>
<p>The function <code>readfmridata</code> can read fMRI data file(s)
stored in ANALYZE format (.img/.hdr files), NIFTI format (.img/.hdr
files or .nii files) or AFNI format (.BRIK/.HEAD files). The reading of
the fMRI data files is done using R package <strong>oro.nifti</strong>
<span class="citation">(Whitcher, Schmid, and Thornton 2011)</span>,
which is loaded when <strong>BHMSMAfMRI</strong> package is loaded.</p>
<p>For illustration, we consider a simulated fMRI dataset stored in
ANALYZE format and provided within the <strong>BHMSMAfMRI</strong>
package. The dataset contains noisy fMRI data collected over a <span class="math inline">\(32 \times 32\)</span> grid of a single axial brain
slice and at 9 timepoints for 3 subjects. The following code illustrates
how the function <code>readfmridata</code> can be used to import the
data from the image files. The simulated dataset is extracted in the
directory <code>fmridata</code> within the R temporary folder.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/nilotpalsanyal/BHMSMAfMRI" class="external-link">BHMSMAfMRI</a></span><span class="op">)</span></span>
<span><span class="va">fpath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, package<span class="op">=</span><span class="st">"BHMSMAfMRI"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/untar.html" class="external-link">untar</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">fpath</span>,<span class="st">"/fmridata.tar"</span><span class="op">)</span>, exdir<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">n</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">grid</span> <span class="op">&lt;-</span> <span class="fl">32</span></span>
<span><span class="va">ntime</span> <span class="op">&lt;-</span> <span class="fl">9</span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/array.html" class="external-link">array</a></span><span class="op">(</span>dim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">n</span>,<span class="va">grid</span>,<span class="va">grid</span>,<span class="va">ntime</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">subject</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">)</span></span>
<span><span class="op">{</span></span>
<span>  <span class="va">directory</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span>,<span class="st">"/fmridata"</span>,<span class="st">"/s0"</span>,<span class="va">subject</span>,<span class="st">"/"</span><span class="op">)</span></span>
<span>  <span class="va">a</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readfmridata.html">readfmridata</a></span><span class="op">(</span><span class="va">directory</span>, format<span class="op">=</span><span class="st">"Analyze"</span>, </span>
<span>  prefix<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"s0"</span>,<span class="va">subject</span>,<span class="st">"_t"</span><span class="op">)</span>, nimages<span class="op">=</span><span class="va">ntime</span>, dim.image<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">grid</span>,<span class="va">grid</span>,<span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">data</span><span class="op">[</span><span class="va">subject</span>,,,<span class="op">]</span> <span class="op">&lt;-</span> <span class="va">a</span><span class="op">[</span>,,<span class="fl">1</span>,<span class="op">]</span></span>
<span><span class="op">}</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">a</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 32 32  1  9</span></span></code></pre>
<p>The above code reads all the data files for all subjects into a 4D
array <code>data</code>. For each subject, the data were generated by
adding Gaussian random noise to the true regression coefficient image
with activation in 3 regions. The positions of 2 activation regions were
varied across subject. The underlying design was a block design. The
true regression coefficient images and the design matrix are also
included in the package and can be read as follows.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">fmridata</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">fmridata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "grid"         "nsubject"     "TrueCoeff"    "DesignMatrix"</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">truecoef</span> <span class="op">&lt;-</span> <span class="va">fmridata</span><span class="op">$</span><span class="va">TrueCoeff</span></span>
<span><span class="va">designmat</span> <span class="op">&lt;-</span> <span class="va">fmridata</span><span class="op">$</span><span class="va">DesignMatrix</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">truecoef</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  3 32 32</span></span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">designmat</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 9 2</span></span></code></pre>
<p>Now, we have <code>truecoef</code> which is an array of dimension (3,
32, 32) containing the true regression coefficients, <code>data</code>
which is an array of dimension (3, 32, 32, 9) containing time series of
noisy observations for all the subjects and <code>designmat</code> which
is the design matrix used to generate the data. Note that, the R package
<strong>neuRosim</strong> <span class="citation">(Welvaert et al.
2011)</span> can be used to generate fMRI data.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="va">n</span><span class="op">)</span>, cex<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">subject</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="va">truecoef</span><span class="op">[</span><span class="va">subject</span>,,<span class="op">]</span>, main<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Subject "</span>,<span class="va">subject</span><span class="op">)</span>, </span>
<span>         col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">heat.colors</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BHMSMAfMRIvignette_files/figure-html/TrueCoef-1.png" alt="True regression coefficient images for the 3 subjects" width="1152"><p class="caption">
True regression coefficient images for the 3 subjects
</p>
</div>
</div>
<div class="section level3">
<h3 id="temporal-modeling-through-glm-glmcoef">Temporal modeling through GLM: <code>glmcoef</code><a class="anchor" aria-label="anchor" href="#temporal-modeling-through-glm-glmcoef"></a>
</h3>
<p>Now, we fit a general linear model to the time series of each voxel
and obtain the estimated regression coefficients for all the regressors
included in <code>designmat</code> by using the function
<code>glmcoef</code> as follows.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">glmmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/glmcoef.html">glmcoef</a></span><span class="op">(</span><span class="va">n</span>, <span class="va">grid</span>, <span class="va">data</span>, <span class="va">designmat</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">glmmap</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "GLMCoefStandardized" "GLMCoefSE"</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">glmmap</span><span class="op">$</span><span class="va">GLMCoefStandardized</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  3 32 32  2</span></span></code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">glmmap</span><span class="op">$</span><span class="va">GLMCoefSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  3 32 32  2</span></span></code></pre>
<p>The output <code>glmmap</code> contains the estimated standardized
regression coefficients and their standard error estimates. From now on,
we focus on the 2<span class="math inline">\(^\text{nd}\)</span>
regressor as the regressor of interest. Figure @ref(fig:GLMCoef),
obtained by the following code, shows the images of its standardized
regression coefficients estimates for the 3 subjects.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">k</span> <span class="op">&lt;-</span> <span class="fl">2</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="va">n</span><span class="op">)</span>, cex<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">subject</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">glmmap</span><span class="op">$</span><span class="va">GLMCoefStandardized</span><span class="op">[</span><span class="va">subject</span>,,,<span class="va">k</span><span class="op">]</span><span class="op">)</span>, col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">heat.colors</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>,</span>
<span>         zlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">6</span><span class="op">)</span>, main<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Subject "</span>,<span class="va">subject</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BHMSMAfMRIvignette_files/figure-html/GLMCoef-1.png" alt="Standardized regression coefficient estimates images for the second regressor for all subjects" width="1152"><p class="caption">
Standardized regression coefficient estimates images for the second
regressor for all subjects
</p>
</div>
</div>
<div class="section level3">
<h3 id="wavelet-transform-of-the-glm-coefficients-waveletcoef">Wavelet transform of the GLM coefficients:
<code>waveletcoef</code><a class="anchor" aria-label="anchor" href="#wavelet-transform-of-the-glm-coefficients-waveletcoef"></a>
</h3>
<p>Next, we apply the discrete wavelet transform to the standardized
regression coefficient images of each subject. The wavelet
transformation is performed by the using the R package
<strong>wavethresh</strong> <span class="citation">(Nason 2013)</span>,
which is loaded when <strong>BHMSMAfMRI</strong> package is loaded.</p>
<p>The function <code>waveletcoef</code> returns the wavelet
coefficients for the selected regressor for all the subjects as a
matrix. Below it is illustrated.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">wavecoefglmmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/waveletcoef.html">waveletcoef</a></span><span class="op">(</span><span class="va">n</span>, <span class="va">grid</span>, <span class="va">glmmap</span><span class="op">$</span><span class="va">GLMCoefStandardized</span><span class="op">[</span>,,,<span class="va">k</span><span class="op">]</span>, </span>
<span>            wave.family<span class="op">=</span><span class="st">"DaubLeAsymm"</span>, filter.number<span class="op">=</span><span class="fl">6</span>, bc<span class="op">=</span><span class="st">"periodic"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">wavecoefglmmap</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "WaveletCoefficientMatrix"</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">wavecoefglmmap</span><span class="op">$</span><span class="va">WaveletCoefficientMatrix</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]    3 1023</span></span></code></pre>
<p>In the wavelet transform, the user can choose the wavelet family (one
of <code>DaubLeAsymm</code> and <code>DaubExPhase</code>), the number of
vanishing moments (<code>filter.number</code>) and the boundary
condition (<code>symmetric</code> or <code>periodic</code>) to be
applied. For fMRI data, we recommend the use of Daubechies least
asymmetric wavelet transform (<code>DaubLeAsymm</code>) with 6 vanishing
moments and periodic boundary condition.</p>
</div>
<div class="section level3">
<h3 id="estimating-the-model-hyperparameters-hyperest">Estimating the model hyperparameters: <code>hyperest</code><a class="anchor" aria-label="anchor" href="#estimating-the-model-hyperparameters-hyperest"></a>
</h3>
<p>The BHMSMA model has six hyperparameters, which are estimated by
their maximum likelihood estimates (MLEs) following an empirical Bayes
approach. We can estimate the hyperparameters by performing
multi-subject analysis or single subject analysis. In multi-subject
analysis, the likelihood function of the hyperparameters is constructed
over all subjects and maximized to obtain their estimates. In single
subject analysis, for each subject, separate likelihood function of the
hyperparameters is constructed and maximized. Hence, for single subject
analysis, for each subject we obtain a set of estimates of the
hyperparameters. Clearly, multi-subject analysis benefits from being
able to borrow strength across subjects and produces more precise
estimates.</p>
<p>The function <code>hyperparamest</code> computes the hyperparameter
estimates and their standard error estimates. The type of analysis must
be specified as <code>analysis="multi"</code> or <code>"single"</code>.
The following code illustrates the use of the function
<code>hyperparamest</code> and the output.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">hyperest</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/hyperparamest.html">hyperparamest</a></span><span class="op">(</span><span class="va">n</span>, <span class="va">grid</span>, <span class="va">wavecoefglmmap</span><span class="op">$</span><span class="va">WaveletCoefficientMatrix</span>, </span>
<span>                           analysis <span class="op">=</span> <span class="st">"multi"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">hyperest</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "hyperparam"    "hyperparamVar"</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparam</span>,<span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1.295 1.193 0.617 0.814 2.242 0.204</span></span></code></pre>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">signif</a></span><span class="op">(</span><span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparamVar</span>,<span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            [,1]       [,2]       [,3]       [,4]       [,5]       [,6]</span></span>
<span><span class="co">## [1,]  5.085e-19  2.002e-34 -5.642e-35 -1.241e-35  1.368e-34  5.295e-35</span></span>
<span><span class="co">## [2,]  2.002e-34  4.319e-19 -1.040e-34  1.143e-35  4.097e-34  2.583e-35</span></span>
<span><span class="co">## [3,] -5.642e-35 -1.040e-34  1.153e-19  5.909e-36 -8.143e-35 -3.115e-35</span></span>
<span><span class="co">## [4,] -1.241e-35  1.143e-35  5.909e-36  2.008e-19 -6.876e-34  1.957e-35</span></span>
<span><span class="co">## [5,]  1.368e-34  4.097e-34 -8.143e-35 -6.876e-34  1.525e-18 -5.394e-36</span></span>
<span><span class="co">## [6,]  5.295e-35  2.583e-35 -3.115e-35  1.957e-35 -5.394e-36  1.265e-20</span></span></code></pre>
<p>From the hyperparameter estimates, we can compute the estimates of
<span class="math inline">\(a_{kl}\)</span>, <span class="math inline">\(b_{kl}\)</span> and <span class="math inline">\(c_{kl}\)</span> <span class="citation">(Sanyal and
Ferreira 2012)</span> for all levels as follows.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">a.kl</span> <span class="op">&lt;-</span> <span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparam</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">*</span> <span class="fl">2</span><span class="op">^</span><span class="op">(</span><span class="op">-</span><span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparam</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span> <span class="op">*</span> <span class="op">(</span><span class="fl">0</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">b.kl</span> <span class="op">&lt;-</span> <span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparam</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span> <span class="op">*</span> <span class="fl">2</span><span class="op">^</span><span class="op">(</span><span class="op">-</span><span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparam</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span> <span class="op">*</span> <span class="op">(</span><span class="fl">0</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">c.kl</span> <span class="op">&lt;-</span> <span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparam</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span> <span class="op">*</span> <span class="fl">2</span><span class="op">^</span><span class="op">(</span><span class="op">-</span><span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparam</span><span class="op">[</span><span class="fl">6</span><span class="op">]</span> <span class="op">*</span> <span class="op">(</span><span class="fl">0</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">a.kl</span>,<span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 1.295 0.566 0.248 0.108 0.047</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="computing-posterior-distribution-of-the-wavelet-coefficients-postmixprob-postwaveletcoef">Computing posterior distribution of the wavelet coefficients:
<code>postmixprob</code>, <code>postwaveletcoef</code><a class="anchor" aria-label="anchor" href="#computing-posterior-distribution-of-the-wavelet-coefficients-postmixprob-postwaveletcoef"></a>
</h3>
<p>Given the values of the hyperparameters, the marginal posterior
distribution of the wavelet coefficients is a mixture of a Gaussian and
a point mass at zero with mixture probabilities <span class="math inline">\(\bar{p}_{iklj}\)</span>. BHMSMA computes <span class="math inline">\(\bar{p}_{iklj}\)</span> values using Newton-Cotes
numerical integration method. The function <code>postmixprob</code>
computes the values <span class="math inline">\(\bar{p}_{iklj}\)</span>
for all subjects and returns in a matrix. The following code illustrates
it.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pkljbar</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/postmixprob.html">postmixprob</a></span><span class="op">(</span><span class="va">n</span>, <span class="va">grid</span>, <span class="va">wavecoefglmmap</span><span class="op">$</span><span class="va">WaveletCoefficientMatrix</span>, </span>
<span>                        <span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparam</span>, analysis <span class="op">=</span> <span class="st">"multi"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pkljbar</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "pkljbar"</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">pkljbar</span><span class="op">$</span><span class="va">pkljbar</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]    3 1023</span></span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">pkljbar</span><span class="op">$</span><span class="va">pkljbar</span><span class="op">[</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>,<span class="fl">4</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##  [1] 0.9868 0.9115 0.9466 0.9095 0.9174 0.9237 0.8922 0.8911 0.9998 0.9599</span></span></code></pre>
<p>Once <span class="math inline">\(\bar{p}_{iklj}\)</span> values are
obtained, the marginal posterior distribution of the wavelet
coefficients are entirely known. With the hyperparameter estimates and
the <span class="math inline">\(\bar{p}_{iklj}\)</span> values, the
function <code>postwaveletcoef</code> computes the posterior mean and
the posterior median of the wavelet coefficients. The following code
illustrates it.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">postwavecoefglmmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/postwaveletcoef.html">postwaveletcoef</a></span><span class="op">(</span><span class="va">n</span>, <span class="va">grid</span>, </span>
<span>                         <span class="va">wavecoefglmmap</span><span class="op">$</span><span class="va">WaveletCoefficientMatrix</span>, </span>
<span>                 <span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparam</span>, <span class="va">pkljbar</span><span class="op">$</span><span class="va">pkljbar</span>, analysis <span class="op">=</span> <span class="st">"multi"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">postwavecoefglmmap</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "PostMeanWaveletCoeff"   "PostMedianWaveletCoeff"</span></span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">postwavecoefglmmap</span><span class="op">$</span><span class="va">PostMeanWaveletCoef</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]    3 1023</span></span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">postwavecoefglmmap</span><span class="op">$</span><span class="va">PostMedianWaveletCoeff</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]    3 1023</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="computing-posterior-mean-of-the-regression-coefficients-postglmcoef">Computing posterior mean of the regression coefficients:
<code>postglmcoef</code><a class="anchor" aria-label="anchor" href="#computing-posterior-mean-of-the-regression-coefficients-postglmcoef"></a>
</h3>
<p>Given the posterior mean of the wavelet coefficients, the function
<code>postglmcoef</code> is used to obtain the posterior means of the
regression coefficients. The following code shows its use.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">postglmmap</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/postglmcoef.html">postglmcoef</a></span><span class="op">(</span><span class="va">n</span>, <span class="va">grid</span>, <span class="va">glmmap</span><span class="op">$</span><span class="va">GLMCoefStandardized</span><span class="op">[</span>,,,<span class="va">k</span><span class="op">]</span>, </span>
<span>         <span class="va">postwavecoefglmmap</span><span class="op">$</span><span class="va">PostMeanWaveletCoef</span>, wave.family<span class="op">=</span><span class="st">"DaubLeAsymm"</span>, </span>
<span>           filter.number<span class="op">=</span><span class="fl">6</span>, bc<span class="op">=</span><span class="st">"periodic"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">postglmmap</span>,vec.len <span class="op">=</span> <span class="fl">3</span>, digits.d <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## List of 1</span></span>
<span><span class="co">##  $ GLMcoefposterior: num [1:3, 1:32, 1:32] -0.36 -0.188 -0.014 0.231 ...</span></span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="va">n</span><span class="op">)</span>, cex<span class="op">=</span><span class="fl">1</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">subject</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">postglmmap</span><span class="op">$</span><span class="va">GLMcoefposterior</span><span class="op">[</span><span class="va">subject</span>,,<span class="op">]</span><span class="op">)</span>, col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">heat.colors</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>,</span>
<span>         zlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">6</span><span class="op">)</span>, main<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Subject "</span>,<span class="va">subject</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BHMSMAfMRIvignette_files/figure-html/PostCoef-1.png" alt="Posterior standardized regression coefficient images for the 3 subjects obtained by BHMSMA" width="1152"><p class="caption">
Posterior standardized regression coefficient images for the 3 subjects
obtained by BHMSMA
</p>
</div>
<p>Figure @ref(fig:PostCoef), obtained by the following code, shows the
images of the posterior standardized regression coefficients for the 3
subjects.</p>
<p>As the true coefficients are known, we can compute the mean squared
error (MSE) using the following code.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">MSE</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">)</span> </span>
<span> <span class="va">MSE</span><span class="op">[</span><span class="va">i</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">truecoef</span><span class="op">[</span><span class="va">i</span>,,<span class="op">]</span><span class="op">/</span><span class="va">glmmap</span><span class="op">$</span><span class="va">GLMCoefSE</span><span class="op">[</span><span class="va">i</span>,,,<span class="fl">2</span><span class="op">]</span><span class="op">)</span>  </span>
<span>                <span class="op">-</span> <span class="fu"><a href="https://rdrr.io/r/base/vector.html" class="external-link">as.vector</a></span><span class="op">(</span><span class="va">postglmmap</span><span class="op">$</span><span class="va">GLMcoefposterior</span><span class="op">[</span><span class="va">i</span>,,<span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">MSE</span>,<span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 446.322 309.666 306.514</span></span></code></pre>
<p>In <span class="citation">(Sanyal and Ferreira 2012)</span>, we show
that our multi-subject methodology performs better than some existing
methodologies in terms of MSE.</p>
</div>
<div class="section level3">
<h3 id="posterior-simulation-and-uncertainty-estimation-postsamples">Posterior simulation and uncertainty estimation:
<code>postsamples</code><a class="anchor" aria-label="anchor" href="#posterior-simulation-and-uncertainty-estimation-postsamples"></a>
</h3>
<p>In order to simulate observations from the posterior distribution of
the regression coefficients, the function <code>postsamples</code> can
be used. The type of analysis must be mentioned. The code below shows
its use.</p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">Postsamp</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/postsamples.html">postsamples</a></span><span class="op">(</span> nsample<span class="op">=</span><span class="fl">50</span>, <span class="va">n</span>, <span class="va">grid</span>, <span class="va">glmmap</span><span class="op">$</span><span class="va">GLMCoefStandardized</span><span class="op">[</span>,,,<span class="va">k</span><span class="op">]</span>,</span>
<span>               <span class="va">wavecoefglmmap</span><span class="op">$</span><span class="va">WaveletCoefficientMatrix</span>,  <span class="va">hyperest</span><span class="op">$</span><span class="va">hyperparam</span>, </span>
<span>               <span class="va">pkljbar</span><span class="op">$</span><span class="va">pkljbar</span>, <span class="st">"multi"</span>, seed<span class="op">=</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">Postsamp</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "samples"       "postdiscovery"</span></span></code></pre>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">Postsamp</span><span class="op">$</span><span class="va">samples</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  3 32 32 50</span></span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">Postsamp</span><span class="op">$</span><span class="va">postdiscovery</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1]  3 32 32</span></span></code></pre>
<p>The argument <code>nsample</code> denotes the number of samples to be
drawn. We can see <code>postsamples</code> returns the posterior samples
and the probabilities of posterior discovery <span class="citation">(Morris et al. 2011)</span> for all the subjects.
Figure @ref(fig:PostDiscovery), obtained by the following code, shows
the posterior discovery images based on the above 50 samples for the 3
subjects.</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="va">n</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">subject</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="va">Postsamp</span><span class="op">$</span><span class="va">postdiscovery</span><span class="op">[</span><span class="va">subject</span>,,<span class="op">]</span>, col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">heat.colors</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>,</span>
<span>   main<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Subject "</span>,<span class="va">subject</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BHMSMAfMRIvignette_files/figure-html/PostDiscovery-1.png" alt="Posterior discovery images for the 3 subjects" width="1152"><p class="caption">
Posterior discovery images for the 3 subjects
</p>
</div>
<p>From the posterior samples, the posterior standard deviations of the
regression coefficients can be computed as follows.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">postsd</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/array.html" class="external-link">array</a></span><span class="op">(</span>dim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">n</span>,<span class="va">grid</span>,<span class="va">grid</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="kw">for</span><span class="op">(</span><span class="va">subject</span> <span class="kw">in</span> <span class="fl">1</span><span class="op">:</span><span class="va">n</span><span class="op">)</span></span>
<span> <span class="va">postsd</span><span class="op">[</span><span class="va">subject</span>,,<span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">Postsamp</span><span class="op">$</span><span class="va">samples</span><span class="op">[</span><span class="va">subject</span>,,,<span class="op">]</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="va">sd</span><span class="op">)</span></span>
<span>                             <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">postsd</span><span class="op">[</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>,<span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##       [,1]  [,2]  [,3]  [,4]  [,5]</span></span>
<span><span class="co">## [1,] 0.466 0.496 0.471 0.677 0.677</span></span>
<span><span class="co">## [2,] 0.655 0.737 0.531 0.677 0.687</span></span>
<span><span class="co">## [3,] 0.657 0.465 0.472 0.435 0.656</span></span>
<span><span class="co">## [4,] 0.656 0.702 0.470 0.581 0.597</span></span>
<span><span class="co">## [5,] 0.571 0.468 0.417 0.532 0.508</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="posterior-group-estimates-postgroupglmcoef">Posterior group estimates: <code>postgroupglmcoef</code><a class="anchor" aria-label="anchor" href="#posterior-group-estimates-postgroupglmcoef"></a>
</h3>
<p>Posterior group coefficients can be obtained by using the function
<code>postgroupglmcoef</code> as follows.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">postgroup</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/postgroupglmcoef.html">postgroupglmcoef</a></span><span class="op">(</span> <span class="va">n</span>, <span class="va">grid</span>, <span class="va">glmmap</span><span class="op">$</span><span class="va">GLMCoefStandardized</span><span class="op">[</span>,,,<span class="va">k</span><span class="op">]</span>, </span>
<span>                               <span class="va">postwavecoefglmmap</span><span class="op">$</span><span class="va">PostMeanWaveletCoef</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">postgroup</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "groupcoef"</span></span></code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">postgroup</span><span class="op">$</span><span class="va">groupcoef</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 32 32</span></span></code></pre>
<p>Figure @ref(fig:PostGroupCoef), obtained by the following code, shows
the posterior group coefficient image for the simulated dataset.</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">1</span><span class="op">)</span>,cex<span class="op">=</span><span class="fl">0.5</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/image.html" class="external-link">image</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">postgroup</span><span class="op">$</span><span class="va">groupcoef</span><span class="op">)</span>,col<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/grDevices/palettes.html" class="external-link">heat.colors</a></span><span class="op">(</span><span class="fl">8</span><span class="op">)</span>,zlim<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>,<span class="fl">6</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure" style="text-align: center">
<img src="BHMSMAfMRIvignette_files/figure-html/PostGroupCoef-1.png" alt="Posterior group regression coefficient image" width="288"><p class="caption">
Posterior group regression coefficient image
</p>
</div>
</div>
<div class="section level3">
<h3 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-AFNI" class="csl-entry">
Cox, R. W. 1996. <span>“AFNI: Software for Analysis and Visualization of
Functional Magnetic Resonance Neuroimages.”</span> <em>Computers and
Biomedical Research</em> 29 (3): 162–73.
</div>
<div id="ref-SPM" class="csl-entry">
Friston, K. J., J. Ashburner, S. J. Kiebel, T. E. Nichols, and W. D.
Penny, eds. 2007. <em>Statistical Parametric Mapping: The Analysis of
Functional Brain Images</em>. Academic Press.
</div>
<div id="ref-BrainVoyager" class="csl-entry">
Goebel, R., F. Esposito, and E. Formisano. 2006. <span>“Analysis of
Functional Image Analysis Contest (FIAC) Data with BrainVoyager QX: From
Single-Subject to Cortically Aligned Group General Linear Model Analysis
and Self-Organizing Group Independent Component Analysis.”</span>
<em>Human Brain Mapping</em> 27 (5): 392–401.
</div>
<div id="ref-morris:et:al:2011" class="csl-entry">
Morris, J. S., V. Baladandayuthapani, R. C. Herrick, P. Sanna, and H.
Gutstein. 2011. <span>“Automated Analysis of Quantitative Image Data
Using Isomorphic Functional Mixed Models, with Application to Proteomic
Data.”</span> <em>Annals of Applied Statistics</em> 5: 894–923.
</div>
<div id="ref-wavethresh" class="csl-entry">
Nason, Guy. 2013. <em><span>“Wavethresh: Wavelets Statistics and
Transforms.”</span></em>
</div>
<div id="ref-sanyal:ferreira:2012" class="csl-entry">
Sanyal, N., and M. A. R. Ferreira. 2012. <span>“Bayesian Hierarchical
Multi-Subject Multiscale Analysis of Functional MRI Data.”</span>
<em>NeuroImage</em> 63 (3): 1519–31.
</div>
<div id="ref-FSL" class="csl-entry">
Smith, S. M., M. Jenkinson, M. W. Woolrich, C. F. Beckmann, T. E. J.
Behrens, H. Johansen-Berg, P. R. Bannister, et al. 2004. <span>“Advances
in Functional and Structural MR Image Analysis and Implementation as
FSL.”</span> <em>NeuroImage</em> 23 (SUPPL. 1): S208–19.
</div>
<div id="ref-neuRosim" class="csl-entry">
Welvaert, Marijke, Joke Durnez, Beatrijs Moerkerke, Geert Verdoolaege,
and Yves Rosseel. 2011. <span>“<span class="nocase">neuRosim</span>: An
<span>R</span> Package for Generating fMRI Data.”</span> <em>Journal of
Statistical Software</em> 44 (10): 1–18.
</div>
<div id="ref-oronifti" class="csl-entry">
Whitcher, Brandon, Volker J. Schmid, and Andrew Thornton. 2011.
<span>“Working with the <span>DICOM</span> and <span>NIfTI</span> Data
Standards in <span>R</span>.”</span> <em>Journal of Statistical
Software</em> 44 (6): 1–28.
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Nilotpal Sanyal.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
